src/backend/executor/execUtils.c

```diff
@@ -2204,6 +2204,17 @@ void mppExecutorFinishup(QueryDesc *queryDesc)
 		estate->es_interconnect_is_setup = false;
 	}
 
+	/*
+	 * If we are finishing a query before all the tuples of the query
+	 * plan were fetched we must call ExecSquelchNode before checking
+	 * the dispatch results in order to tell the nodes below we no longer
+	 * need any more tuples.
+	 */
+	if (!estate->es_got_eos)
+	{
+		ExecSquelchNode(queryDesc->planstate);
+	}
```

Comment:
I think we should exclude executor processes from running `ExecSquelchNode` as was done before. Because:

1. `queryDesc->planstate` may point to plan node for another slice in case of `execute_pruned_plan` is off.
2. otherwise, `queryDesc->planstate` will point to Motion on top of slice. But ExecSquelchNode for motion nodes doesn't walk to children nodes anyway, because it must be called only for Motions in the lower part of slice (Motion receivers) to send a stop command to senders that will call ExecSquelchNode for their children.

I didn't find any cases when `estate->es_got_eos` is left in false on executors, but I think it would be nice to strict running on executors explicitly.

Reply:
done

Reply:
Can you prove that dispatcherState is filled on executors to avoid this section?

Reply:
redesigned